### PR TITLE
Add tenderly hardhat config and deployment script

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Adds a new deploy script that verifies the contracts selected on Tenderly which will allow us to get full traces and debug easier

It can be run with:
```
npx hardhat deploy --network polygonTestnet --tags Tenderly
```

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: deploy/31_tenderly.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+declare var global: any;
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from '@holographxyz/hardhat-deploy-holographed/types';
+import { NetworkType, networks } from '@holographxyz/networks';
+import { SuperColdStorageSigner } from 'super-cold-storage-signer';
+
+// ts-ignore because this is not a hardhat plugin
```
Comment (alexanderattar) [inline deploy/31_tenderly.ts]: Remove this

File: hardhat.config.ts [hunk 2]
```diff
@@ -374,6 +380,27 @@ const config: HardhatUserConfig = {
...
+  tenderly: {
+    project: 'holograph',
+    username: 'attar',
+    privateVerification: false,
+    forkNetwork: '<fork-chain-id>',
+  },
 };
 
+// Allow hardhat to use short network names
+function mapNetworkKeysByShortKey(networks: Networks) {
```
Comment (sogoiii) [inline hardhat.config.ts:392]: @alexanderattar Given that this is used in several spots, should we move this into a file that we export? Would be good to not have duplication.

Comment (alexanderattar) [reply hardhat.config.ts:392]: it's actually only being used in hardhat.config at the moment

Comment (sogoiii) [reply hardhat.config.ts:392]: Ah ok, i see. I confused it with `mapFullKeyToShortKey`

**Review by sogoiii — APPROVED** [review]
👍 this is a great tool.

**Pull request merged**